odoo/api.py

```diff
@@ -557,7 +557,9 @@ def company(self):
                 if any(cid not in user_company_ids for cid in company_ids):
                     raise AccessError(_("Access to unauthorized or invalid companies."))
             return self['res.company'].browse(company_ids[0])
-        return self.user.company_id
+        # prefetch_fields is necessary during module installation
+        # because the res_users table may not have all prescribed columns.
+        return self.user.with_context(prefetch_fields=False).company_id
```

Comment:
Ouch!  This one adds a query (see tests on mail).

Maybe a try/except?
```python
try:
    return self.user.company_id
except TheExceptionRaisedByPsycopg2:
    return self.user.with_context(prefetch_fields=False).company_id
```

Reply:
Another solution is to add the `with_context` inside `get_model_defaults`:
```python
params = [model_name, self.env.uid, self.with_context(prefetch_fields=False).env.company.id]
```

Reply:
The second solution looks better to me, as it will be more specific to the `ir.default` case...

Reply:
Well, it needs a mute_logger and a savepoint, not sure we want to go this far :thinking:

Reply:
Yes, but then we have to use the with_context in the orm_cache values also ^^.